### DISCUSSION TITLE
Standardizing colors and having better visibility for links on chat

### DISCUSSION THREAD

**Discussion by Nukrid** [ppy/osu#23145] [Ideas]
Pretty simple suggestion. I have talked to Ephemeral earlier about this, shared thoughts with both #osu regulars and team members, and lots of people say that the current color scheme on chat is a bit confusing.

![image](https://user-images.githubusercontent.com/61538813/230310459-aa3a2021-f17c-4781-8c0d-def752c86da2.png)

Currently, having all those sorts of colors makes visibility sort of a problem in my opinion, not only it feels off (comparing to Stable), but it feels a bit overwhelming, having to look through all those different colors and trying to sort people out. What I would suggest is that the colors are standardized like we have on Stable (Pale Yellow and Yellow, but could be any other alternative that may look good) for normal users. Particularly, I don't have any visibility issues regarding team members, as they are inside a neat box that highlights them.

For links, we could have them inside a box as well, or at least having a color that draws more attention, because right now, links almost look like any other message, so they don't draw any attention, especially when we are looking for people throwing some weird stuff on chats.

I believe the colors may be something that we can get used to, but links really should get more visibility, at least.

**Comment by peppy** [comment]
The username colours are intended to let you follow individual users without reading their full username. Is your issue the fact that this logic is existing, or just that the colour palette itself sucks? I agree with the latter.

@arflyte comments on the link colour?

**Reply by Nukrid** [reply]
The colors could be a bit easier on the eyes (like the pink one in the screenshot, it is a bit too strong), that is the main issue for me. I also get a bit sidetracked about having them different, but I can surely get used to it.

**Reply by peppy** [reply]
@arflyte still with us?

**Comment by arflyte** [comment]
The idea of having multiple colours for user was to ensure that we know who's talking at a single glance. But seeing that Discord is the popular chat app now, it can be confusing now. We can tone down amount of colours available, and maybe just give colours to usergroups (like Discord).

As for the link colours, it's part of the main colour palette (links are a shade of main active colour). So far our colour palette is monotonous, and I'm currently in the process of adding complementary colours to it to make it less.. "monotonous". I believe this will fix the link problem. Temporarily, we can change the links to blue (like Discord).

Putting links in a box is too distracting, so let's not do that.

**Reply by peppy** [reply]
Can you please provide a suggestion for a colour palette for chat names? Something easier on the eyes.

**Comment by bdach** [comment]
I'm gonna necro this as it's vaguely related. 

With daily challenge release I'm seeing users complain about their random-assigned username colours and wanting to be able to customise them. Not sure how to provide that, but it is a frequent piece of feedback.

**Reply by Stoppedpuma** [reply]
I saw the idea of having it be a supporter feature a while back, maybe an implementation similar to  how profile colour hue can be customised would work?
